src/ORM/Query.php

```diff
@@ -1132,8 +1132,9 @@ protected function _execute(): ResultSetInterface
         }
 
         $statement = $this->getEagerLoader()->loadExternal($this, $this->execute());
+        $resultset = new ResultSet($this, $statement->fetchAll('assoc') ?: []);
```

Comment:
Yes I don't think it will matter. In general if memory usage becomes an issue for someone then they should pull lesser data or use the db layer directly instead of the ORM.